pkg/kubelet/images/image_gc_manager.go

```diff
@@ -171,30 +171,30 @@ func NewImageGCManager(runtime container.Runtime, statsProvider StatsProvider, r
 
 func (im *realImageGCManager) Start() {
 	go wait.Until(func() {
-		// Initial detection make detected time "unknown" in the past.
-		var ts time.Time
-		if im.initialized {
-			ts = time.Now()
-		}
-		_, err := im.detectImages(ts)
-		if err != nil {
-			klog.Warningf("[imageGCManager] Failed to monitor images: %v", err)
-		} else {
-			im.initialized = true
-		}
-	}, 5*time.Minute, wait.NeverStop)
-
-	// Start a goroutine periodically updates image cache.
-	// TODO(random-liu): Merge this with the previous loop.
-	go wait.Until(func() {
+		counter := 0
 		images, err := im.runtime.ListImages()
 		if err != nil {
 			klog.Warningf("[imageGCManager] Failed to update image list: %v", err)
 		} else {
 			im.imageCache.set(images)
 		}
-	}, 30*time.Second, wait.NeverStop)
+		counter++
 
+		if counter%10 == 0 {
```

Comment:
won't `counter` always be `1` on this line?  `counter` is defined as 0 at the top of the function, and then incremented.

Reply:
counter should be defined outside the goroutine

Once @Random-Liu confirms the refactor, I will update the PR.